packages/interbit-covenant-tools/src/rootCovenant/covenants/manifestCovenant/applyJoinChanges.js

```diff
@@ -0,0 +1,170 @@
+const _ = require('lodash')
+const {
+  startConsumeState,
+  startProvideState,
+  authorizeReceiveActions,
+  authorizeSendActions
+} = require('interbit-covenant-utils')
+const {
+  redispatch,
+  actionTypes: coreActionTypes
+} = require('../../../coreCovenant')
+const {
+  stopConsumeState,
+  stopProvideState,
+  revokeSendActions,
+  revokeReceiveActions
+} = require('../../../coreCovenant/helpers/revokeJoin')
+const { JOIN_TYPES } = require('../../../constants')
+
+const applyJoinChanges = (state, newManifest) => {
+  let nextState = state
+
+  const newJoinActions = manifestJoinsToActions(newManifest)
+  if (!state.manifest) {
+    for (const joinAction of newJoinActions) {
+      nextState = redispatch(nextState, joinAction)
+    }
+
+    return nextState
+  }
+
+  const currManifest = !state.manifest.chainId
+    ? Object.values(state.manifest.manifest)[0]
```

Comment:
Why negate condition? This would be easier to read as:
```
 const currManifest = state.manifest.chainId
    ? state.manifest
    : Object.values(state.manifest.manifest)[0]
```